reference/cd-gitlab-ci.md

```diff
@@ -0,0 +1,183 @@
+---
+title: GitLab CI
+---
+
+This page details how to use [GitLab CI](https://about.gitlab.com/features/gitlab-ci-cd/) to manage deploying
+staging and production stacks based on commits to specific Git branches. This is sometimes
+referred to as Push-to-Deploy.
+
+Pulumi doesn't require any particular arrangement of stacks or workflow to work in a
+continuous integration / continuous deployment system. So the steps described here can be
+altered to fit into any existing type of deployment setup.
+
+## Prerequisites
+- An account on https://app.pulumi.com and that you have created a new project.
+  - As of this writing, we support logins only using GitHub identity.
+  - This just means you will sign-in using your GitHub credentials.
+  - However, pulumi can be run from anywhere and your infrastrucutre code itself can be hosted anywhere.
+- The latest CLI. Installation instructions are [here](https://pulumi.io/quickstart/install.html).
+- A bare repo and set the remote URL to be your GitLab repo.
+
+## Stack and Branch Mappings
+The scripts below act on a hypothetical stack: `acme/product-catalog-service-stack`.
+You can create a stack by running `pulumi stack init`.
+The source code for the stack is in a repository in GitLab and uses `TypeScript` as the language.
+
+**Note**: The names used above are purely for demonstration purposes only.
+You may choose a naming convention that best suits your organization.
+
+Alternatively, you can also run `pulumi new [template]` to create a template project.
+Learn more [here](https://pulumi.io/reference/cli/pulumi_new.html).
+
+## GitLab CI Runners
+
+## Protected Branches
+In order to prevent abuse of protected resources, as well as some sensitive information used
+by your repository, GitLab has the concept of [Protected Branches and Tags](https://gitlab.com/help/user/project/protected_branches.md).
+
+If you are running `pulumi` from any branch other than the `master` branch,
+you are likely to hit an error that the `PULUMI_ACCESS_TOKEN` 
+environment variable (introduced later in this document) cannot be accessed.
+You can fix this by specifying a wildcard regex to allow specific branches to
```

Comment:
> by specifying a wildcard regex to allow specific branches to be able to access the secret environment variables.

Where in the GitLab CI would I go to enable that? What would the correct "wildcard regex" look like?

Reply:
The link to Protected Branches and Tags is mentioned in the paragraph before this one. It has screenshots and an example regex as well.